Lib/sqlite3/test/dbapi.py

```diff
@@ -555,6 +555,10 @@ def test_last_row_id_insert_o_r(self):
         ]
         self.assertEqual(results, expected)
 
+    def test_same_query_in_multiple_cursors(self):
+        cursors = [self.cx.execute("select 1") for _ in range(3)]
+        for cu in cursors:
+            self.assertEqual(cu.fetchall(), [(1,)])
```

Comment:
This test exercises this branch:
https://github.com/python/cpython/blob/006fd869e4798b68e266f5de89c83ddb531a756b/Modules/_sqlite/cursor.c#L536-L542